### PR TITLE
Add snitch module

### PR DESCRIPTION
**Pull request by avranju**

### CODE REVIEW

File: edge-modules/snitch/src/report.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+use libflate::finish::AutoFinish;
+use libflate::gzip::Encoder as GzipEncoder;
+use tar::{Builder as TarBuilder, Header as TarHeader};
+
+use error::Result;
+use influx::QueryResults;
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Interval {
```
Comment (dsajanice) [inline edge-modules/snitch/src/report.rs]: Are we planning to add the missing message numbers?

File: edge-modules/snitch/src/settings.yaml [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+alert:
+  host: http://localhost/
+  path: /workflows/id/triggers/manual/paths/invoke
+  query:
+    api-version: 2016-10-01
+    sp: /triggers/manual/run
+    sv: 1.0
+    sig: password
+influx_url: http://influxdb/
 influx_db_name: eh
```
Comment (dsajanice) [inline edge-modules/snitch/src/settings.yaml]: default db name can be metricsdatabase

File: edge-modules/snitch/src/settings.yaml [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
-influx_url: http://localhost/
+alert:
+  host: http://localhost/
+  path: /workflows/id/triggers/manual/paths/invoke
+  query:
+    api-version: 2016-10-01
+    sp: /triggers/manual/run
+    sv: 1.0
+    sig: password
+influx_url: http://influxdb/
```
Comment (dsajanice) [inline edge-modules/snitch/src/settings.yaml]: default influx url should be http://influxdb:8086

File: edge-modules/snitch/Cargo.toml [hunk 4]
```diff
@@ -6,14 +6,20 @@ publish = false
...
+hex = "0.3"
 http = "0.1"
 hyper = "0.12"
+hyper-tls = "0.3.0"
+libflate = "0.1"
 serde = "1.0"
 serde_derive = "1.0"
 serde_json = "1.0"
 serde_yaml = "0.7"
+tar = "0.4"
```
Comment (aribeironovaes) [inline edge-modules/snitch/Cargo.toml]: Reminder to add to open source the modules we don't have. :) Sorry.

File: edge-modules/snitch/src/report.rs (hunk 1 above)
Comment (avranju) [reply edge-modules/snitch/src/report.rs]: The analyzer does not return that at this time. Is that useful? Shouldn't just the start/end times be sufficient to help debug issues?

File: edge-modules/snitch/src/settings.yaml (hunk 2 above)
Comment (avranju) [reply edge-modules/snitch/src/settings.yaml]: Sure.

File: edge-modules/snitch/src/settings.yaml (hunk 3 above)
Comment (avranju) [reply edge-modules/snitch/src/settings.yaml]: Done

File: edge-modules/snitch/Cargo.toml (hunk 4 above)
Comment (avranju) [reply edge-modules/snitch/Cargo.toml]: Yeah, already have a task in the PBI for this.

File: edge-modules/snitch/src/report.rs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MessageAnalysis {
+    module_id: String,
+    status_code: u16,
+    status_message: String,
+    received_messages_count: u64,
+    last_message_received_at: DateTime<Utc>,
+    missing_messages_between: Vec<Interval>,
+}
```
[the remark this answers is not in the document]
Comment (avranju) [reply edge-modules/snitch/src/report.rs]: Good catch. Fixed.

**Comment by avranju** [comment]
OK, so the reason why the builds are not being triggered is because I created the PR from my fork. Closing this PR so I can do it again off of a branch in this repo. 🤷‍♂️

**Pull request closed without merging**